Automattic/woocommerce-payments
# Remove Charge request from Transaction Details page

This is the first part of the fix for #353.

#### Changes proposed in this Pull Request



- Add missing fields to WC_Payments_API_Charge
  - Those fields will be used to populate the transactions details page, removing the requirement of doing 2 requests to fetch all the necessary data
- Remove `Charge` request from `usePaymentIntentWithChargeFallback` hook
- Fix "Payment method" block style

<details>
<summary>Payment method block changes</summary>

<table>
<tr>
<td>
Before
</td>
</tr>
<tr>
<td>
<img width="1039" alt="Before" src="https://user-images.githubusercontent.com/16882226/184360416-c3d1818a-d59a-4dd9-a007-e3c6ca92c45f.png">
</td>
</tr>
<tr>
<td>
After
</td>
</tr>
<tr>
<td>
<img src="https://user-images.githubusercontent.com/16882226/184360473-9858b153-2f60-4be4-86ab-adf21bf31fba.png" alt="After" />
</td>
</tr>
</table>
</details>







#### Testing instructions




* Sync your `woocommerce-payments-server` to 2339-gh-Automattic/woocommerce-payments-server.
* Perform an order on your test shop
  - Open **Payments > Transactions**
  - Click on the transaction that you have just made
  - It should perform a request to `/wp-json/wc/v3/payments/payment_intents/pi_*` and load the transaction details page without requiring a `/charges` request.
* The unit tests should be passing

-------------------


**Post merge**
